tools/performance-server/shared/src/main/kotlin/org/jetbrains/influxdb/InfluxDB.kt

```diff
@@ -0,0 +1,315 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+package org.jetbrains.influxdb
+
+import kotlin.reflect.KProperty
+import kotlin.js.Promise            // TODO - migrate to multiplatform.
+import kotlin.js.json               // TODO - migrate to multiplatform.
+import org.jetbrains.report.json.*
+
+// Connector with InfluxDB.
+object InfluxDBConnector {
```

Comment:
Why singleton? Cannot we have multiple connections actually?
Even if so, I'd do logic in class and have separate instance.

Reply:
Any stateful singleton is blocker to composition, so I usually avoid those when could.